src/custom-elements.js

```diff
@@ -562,7 +573,7 @@ let Deferred;
           const node = walker.currentNode;
           if (node[_upgradedProp] && node[_attachedProp]) {
             node[_attachedProp] = false;
-            const definition = this._definitions.get(node.localName);
+            const definition = getDefinition(this._definitions, node);
```

Comment:
FWIW the native implementation in Chrome does this differently: once a definition is associated with an element (at creation or upgrade) we store a pointer directly from the element to the definition.

Given that getDefinition is more complicated now, it might be worth revisiting the space/speed tradeoff.

I note that you're storing "is" on the element. If you wanted to save space you could switch the storage space for "is" to point to the full definition once you have it. If you ever happened to need is (I don't think you do; cloning goes back to the attribute value for example) you could retrieve it from the definition.